tuichain/api/views/loans.py

```diff
@@ -348,8 +359,20 @@ def get_personal_loan_requests(request):
     user = request.user
 
     loan_list = Loan.objects.filter(student=user)
-
-    result = [obj.to_dict() for obj in loan_list]
+    result = np.empty(shape=loan_list.count(), dtype=object)
```

Comment:
Why add a dependency on numpy just to create a list of placeholder objects? One possibility would be to start with an empty list and using `.append()` to append to it as needed. Ditto for other occurrences.